tests/Wallabag/CoreBundle/Helper/DownloadImagesTest.php

```diff
@@ -240,44 +236,27 @@ public function testEnsureOnlyFirstOccurrenceIsReplaced()
 
     public function testProcessSingleImageWithSvg()
     {
-        $httpMockClient = new HttpMockClient();
-        $httpMockClient->addResponse(new Response(200, ['content-type' => 'image/svg+xml'], file_get_contents(__DIR__ . '/../fixtures/modal-content.svg')));
+        $mockHttpClient = new MockHttpClient([new MockResponse(file_get_contents(__DIR__ . '/../fixtures/modal-content.svg'), ['response_headers' => ['Content-Type: image/svg+xml']])]);
 
         $logHandler = new TestHandler();
         $logger = new Logger('test', [$logHandler]);
 
-        $download = new DownloadImages($httpMockClient, sys_get_temp_dir() . '/wallabag_test', 'http://wallabag.io/', $logger);
+        $download = new DownloadImages($mockHttpClient, sys_get_temp_dir() . '/wallabag_test', 'http://wallabag.io/', $logger);
         $res = $download->processSingleImage(123, 'modal-content.svg', 'http://imgur.com/gallery/WxtWY');
 
         $this->assertStringContainsString('/assets/images/9/b/9b0ead26/400e29f9.svg', $res);
     }
 
     public function testProcessSingleImageWithBadSvg()
     {
-        $httpMockClient = new HttpMockClient();
-        $httpMockClient->addResponse(new Response(200, ['content-type' => 'image/svg+xml'], file_get_contents(__DIR__ . '/../fixtures/unnamed.png')));
+        $mockHttpClient = new MockHttpClient([new MockResponse(file_get_contents(__DIR__ . '/../fixtures/unnamed.png'), ['response_headers' => ['Content-Type: image/svg+xml']])]);
 
         $logHandler = new TestHandler();
         $logger = new Logger('test', [$logHandler]);
 
-        $download = new DownloadImages($httpMockClient, sys_get_temp_dir() . '/wallabag_test', 'http://wallabag.io/', $logger);
+        $download = new DownloadImages($mockHttpClient, sys_get_temp_dir() . '/wallabag_test', 'http://wallabag.io/', $logger);
         $res = $download->processSingleImage(123, 'modal-content.svg', 'http://imgur.com/gallery/WxtWY');
 
         $this->assertFalse($res);
     }
-
-    public function testFollowRedirection()
-    {
-        $httpMockClient = new HttpMockClient();
-        $httpMockClient->addResponse(new Response(301, ['content-type' => 'image/png', 'location' => '/final-path.png']));
-        $httpMockClient->addResponse(new Response(200, ['content-type' => 'image/png'], file_get_contents(__DIR__ . '/../fixtures/unnamed.png')));
-
-        $logHandler = new TestHandler();
-        $logger = new Logger('test', [$logHandler]);
-
-        $download = new DownloadImages($httpMockClient, sys_get_temp_dir() . '/wallabag_test', 'http://wallabag.io/', $logger);
-        $res = $download->processSingleImage(123, '', 'https://example.com/unnamed.png');
-
-        $this->assertStringContainsString('/assets/images/9/b/9b0ead26/66953334.png', $res, "Fetch client didn't follow the HTTP redirection");
-    }
```

Comment:
removed because [HttpClient always follow redirection](https://symfony.com/doc/5.x/http_client.html#redirects), and so this test become like testing the framework.
also, it can't work this way as if I force the first response to be a redirect, it will be the one DOwnloadImages service will receive and so will fail.